plugin/resurrect/fuzzy_loader.lua

```diff
@@ -57,32 +77,375 @@ function pub.fuzzy_load(window, pane, callback, opts)
+
+					-- Calculating the maximum file length
+					max_length = math.max(max_length, utf8len(file) + fmt_cost[type])
+
+					-- collecting all relevant information about the file
+					local fmt = opts[string.format("fmt_%s", type)]
+					table.insert(files, {
+						id = type .. utils.separator .. file,
+						filename = file,
+						epoch = epoch,
+						fmt = fmt,
+						type = type,
+					})
+					-- end
+				end
+			end
+		end
+
+		-- simulate static values for `format_label`
+		local min_filename_len = opts.min_filename_size or 10 -- minimum size of the filename to remain decypherable
+		local str_pad = opts.name_truncature or "..."
+		local pad_len = utf8len(str_pad)
+		-- Format the label given the available screen width, starting with the padding and then reducing the
+		-- filename itself up to a minimum length, then the date can be reduced as a last resort but otherwise
+		-- nothing else can be done because the screen is too small
+		---@param win_width number width of the window
+		---@param file table file package with all necessary information
+		---@return string
+		local function format_label(win_width, file)
+			local label = {
+				filename_raw = "",
+				filename_len = 0,
+				separator = "",
+				padding_raw = "",
+				padding_len = 0,
+				name_raw = "",
+				name_fmt = "",
+				name_len = 0,
+				date_raw = "",
+				date_fmt = "",
+				date_len = 0,
+			}
+			local result = ""
+			-- fill raw values and run a dry run of the formatting to measure the resulting length
+			label.filename_raw = file.filename
+			label.filename_len = utf8len(label.filename_raw)
+			if opts.show_state_with_date then
+				label.separator = " "
+				if utf8len(file.filename) < max_length then
+					label.padding_raw = string.rep(".", max_length - utf8len(file.filename) - 1)
+					label.padding_len = utf8len(label.padding_raw)
+				end
+				label.date_raw = file.date
+				if opts.fmt_date then
+					label.date_fmt = opts.fmt_date(label.date_raw)
+					label.date_len = utf8len(strip_format(label.date_fmt))
+				else
+					label.date_fmt = label.date_raw
+					label.date_len = utf8len(label.date_fmt_fmt)
+				end
+			end
+			label.name_raw = label.filename_raw .. label.separator .. label.padding_raw .. label.separator
+			if file.fmt then
+				label.name_fmt = file.fmt(label.name_raw)
+				label.name_len = utf8len(strip_format(label.name_fmt))
+			else
+				label.name_fmt = label.name_raw
+				label.name_len = utf8len(label.name_fmt)
+			end
+
+			-- check the overall width against the available width
+			local width = label.name_len + label.date_len + 4
+			-- `oversize` is the number of character we should remove
+			local oversize = math.max(0, width - win_width)
+
+			if oversize == 0 then
+				-- No oversize for this line, thus we keep it as is (though it shouldn't occur when we use this function)
+				return label.name_fmt .. label.date_fmt
+			else
+				-- we need to save a `oversize` character; first check if the padding can be used
```

Comment:
This is a bit hard to understand, maybe something like this comment instead
```suggestion
				-- we need to remove characters; first check if the padding can be used
```